src/extensions/timeseries/TimeWidgetNew/TimeWidgetNew.tsx

```diff
@@ -202,12 +208,12 @@ const TimeWidgetNew = (props: Props) => {
     const [prevCurrentTime, setPrevCurrentTime] = useState<number | null>(null)
     const [prevTimeRange, setPrevTimeRange] = useState<{min: number, max: number} | null>(null)
 
-    const mainLayer = useCanvasWidgetLayer(createMainLayer)
-    const timeAxisLayer = useCanvasWidgetLayer(createTimeAxisLayer)
-    const panelLabelLayer = useCanvasWidgetLayer(createPanelLabelLayer)
-    const cursorLayer = useCanvasWidgetLayer(createCursorLayer)
+    const mainLayer = useLayer(createMainLayer)
+    const timeAxisLayer = useLayer(createTimeAxisLayer)
+    const panelLabelLayer = useLayer(createPanelLabelLayer)
+    const cursorLayer = useLayer(createCursorLayer)
 
-    const layers = useCanvasWidgetLayers([mainLayer, timeAxisLayer, panelLabelLayer, cursorLayer])
+    const allLayers = useLayers([mainLayer, timeAxisLayer, panelLabelLayer, cursorLayer])
```

Comment:
This is where it was important to use useLayers so that the list object is persistent between calls when the individual layers have not changed.